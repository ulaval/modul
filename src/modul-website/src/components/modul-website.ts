import { ModulVue } from '@ulaval/modul-components/dist/utils/vue/vue';

export class ModulWebsite extends ModulVue {

}
