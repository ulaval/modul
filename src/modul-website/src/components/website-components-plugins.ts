import Vue, { PluginObject } from 'vue';
import { MWCard, MWCARD_NAME } from './card/card';
import { COLOR_NAME, MColor } from './color/color';
import { COMPONENT_API_NAME, MComponentApi } from './component-api/component-api';
import { DEMO_NAME, MDemo } from './demo/demo';
import { DO_NAME, MDo } from './do/do';
import { DONT_NAME, MDont } from './dont/dont';
import { MDynamicTemplate } from './dynamic-template/dynamic-template';
import { MWExpandablePanel } from './expendable-panel/expandable-panel';
import { GO_NAME, MGo } from './go/go';
import { MWHeader, MWHEADER_NAME } from './header/header';
import { HIGHLIGHT_NAME, MHighlight } from './highlight/highlight';
import { ICON_GALLERY_NAME, MIconGallery } from './icon-gallery/icon-gallery';
import { MWLogo, MWLOGO_NAME } from './logo/logo';
import { MARKDOWN_NAME, MMarkdown } from './markdown/markdown';
import { MWMegaMenu, MWMEGAMENU_NAME } from './mega-menu/mega-menu';
import { MWPanel, MWPANEL_NAME } from './panel/panel';
import { MPreview, PREVIEW_NAME } from './preview/preview';
import SliderPlugin from './slider/slider';

const WebsiteComponentsPlugin: PluginObject<any> = {
    install(v): void {
        Vue.component(COLOR_NAME, MColor);
        Vue.component(DEMO_NAME, MDemo);
        Vue.component(DO_NAME, MDo);
        Vue.component(DONT_NAME, MDont);
        Vue.component(GO_NAME, MGo);
        Vue.component(MARKDOWN_NAME, MMarkdown);
        Vue.component(PREVIEW_NAME, MPreview);
        Vue.component(MWHEADER_NAME, MWHeader);
        Vue.component(MWMEGAMENU_NAME, MWMegaMenu);
        Vue.component(MWLOGO_NAME, MWLogo);
        Vue.component(MWCARD_NAME, MWCard);
        Vue.component(MWPANEL_NAME, MWPanel);
        Vue.component(HIGHLIGHT_NAME, MHighlight);
        Vue.component(ICON_GALLERY_NAME, MIconGallery);
        Vue.component(COMPONENT_API_NAME, MComponentApi);
        Vue.component('mw-dynamic-template', MDynamicTemplate);
        Vue.component('mw-expandable-panel', MWExpandablePanel);
        Vue.component('mw-expandable-panel', MWExpandablePanel);
        Vue.use(SliderPlugin);
    }
};

export default WebsiteComponentsPlugin;
