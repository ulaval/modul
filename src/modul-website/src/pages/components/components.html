<div class="mw-components">
    <m-expandable-layout class="mw-components__body"
                         :open="showMenu"
                         panel-width="320px">
        <div class="mw-components__page">
            <m-icon-button class="mw-components__menu-button"
                           @click="toggleMenu()"
                           :icon-name="showMenu ? 'm-svg__close-clear' : 'm-standard-edit'"
                           skin="light"
                           icon-size="20px">
            </m-icon-button>

            <div class="mw-components__content">
                <mw-component-details :component="component"></mw-component-details>
            </div>
        </div>
        <template slot="panel">
            <div class="mw-components__sidemenu">

                <div class="mw-components__search">
                    <m-searchfield></m-searchfield>
                </div>
                <div class="mw-components__sidemenu-body">
                    <div class="mw-components__sidemenu-title">
                        <m-i18n k="website:components-ui"></m-i18n>
                    </div>
                    <m-accordion skin="plain"
                                 icon-position="right"
                                 icon-size="small"
                                 v-for="categoryId in componentCategoryIds"
                                 :key="categoryId"
                                 :open="urlMatchPath(categoryId)">
                        <div class="mw-components__sidemenu-sub-title"
                             slot="header">
                            <m-i18n :k="'categories:'+categoryId"></m-i18n>
                        </div>
                        <div>
                            <div class="mw-components__sidemenu-item"
                                 v-for="component in getCategoryComponents(categoryId)"
                                 :key="component.url">
                                <router-link :to="getRouterIndex(component.url)"
                                             @click.native="onSideMenuSelection()"><span>
                                        {{component.name}}
                                    </span>
                                    <span v-if="component.beta">
                                        <span class="modul-logo__beta mw-components__beta">
                                            <m-i18n k="website:beta"></m-i18n>
                                        </span>
                                    </span>
                                </router-link>
                            </div>
                        </div>
                    </m-accordion>
                </div>
            </div>
        </template>
    </m-expandable-layout>
</div>
