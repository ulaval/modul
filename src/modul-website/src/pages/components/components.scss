@import '../../styles/commons';

.mw-component-details {
    width: 100%;

    &__component-navigation {
        display: flex;
        margin-top: 54px;
        color: $m-color--interactive-light;
    }

    &__component-navigation-next,
    &__component-navigation-prev {
        line-height: 24px;
        position: relative;
        padding: 1rem;
        flex-grow: 1;
        border: 1px solid;
        text-align: center;
        text-decoration: none;
        font-weight: $m-font-weight--semi-bold;
    }

    &__component-navigation-next--icon {
        position: absolute;
        right: 16px;
    }

    &__component-navigation-prev--icon {
        position: absolute;
        left: 16px;
    }
}
