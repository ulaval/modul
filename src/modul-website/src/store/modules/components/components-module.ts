import { Getter, ModuleBase, Mutation } from '@ulaval/modul-vuex/dist/vuex-annotations';
import { PluginObject } from 'vue';
import { Store } from 'vuex';

declare module 'vue/types/vue' {
    interface Vue {
        $componentsModule: ComponentsModule;
    }
}

export interface ComponentsModuleState {
    components: { [k: string]: any | undefined };
    version: string;
}

export class ComponentsModule extends ModuleBase<ComponentsModuleState> {
    constructor(store: Store<any>, moduleName: string = 'components') {
        super(moduleName, {
            components: {},
            version: ''
        }, store);
    }

    @Mutation()
    public setVersion(version: string): void {
        this.state.version = version;
    }

    @Getter()
    public getVersion(): string {
        return this.state.version;
    }
}

export const ComponentsModulePlugin: PluginObject<any> = {
    install(v, options): void {
        if (options) {
            v.prototype.$componentsModule = new ComponentsModule(options.store);
        }

    }
};
