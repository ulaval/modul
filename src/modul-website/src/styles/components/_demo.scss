.modul-demo {
    &__preview {
        div:first-child > :first-child {
            margin-top: 0;
        }
    }
}
