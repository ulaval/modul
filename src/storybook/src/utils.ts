export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
