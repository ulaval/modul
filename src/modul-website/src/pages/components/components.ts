
import { MWExpandablePanel } from '@/components/expendable-panel/expandable-panel';
import { ModulWebsite } from '@/components/modul-website';
import { ComponentMeta } from '@/content/components.meta.loader';
import _ from 'lodash';
import Vue from 'vue';
import Component from 'vue-class-component';
import { Prop, Watch } from 'vue-property-decorator';
import { MWComponent } from './component/component-details';
import WithRender from './components.html?style=./components.scss';

@WithRender
@Component({
    components: { 'mw-component-details': MWComponent }
})
export class MWComponentsPage extends ModulWebsite {

    @Prop()
    component: ComponentMeta;

    $refs: {
        expandablePanel: MWExpandablePanel;
        prevLink: Vue;
        nextLink: Vue;
    };

    fullPath: string;

    get componentCategoryIds(): string[] {
        return _.sortBy(this.$meta.categories, name => _.deburr(this.$i18n.translate(`categories:${name}`)));
    }

    get componentsNavigationOrder(): ComponentMeta[] {
        let components: ComponentMeta[] = [];
        this.componentCategoryIds.forEach(category => {
            this.$meta.componentState[category].forEach((componentMeta: ComponentMeta) => {
                components.push(componentMeta);
            });
        });
        return components;
    }

    get currentComponentNavigationIndex(): number {
        return _.findIndex(this.componentsNavigationOrder, (c: ComponentMeta) => {
            return this.component.name === c.name;
        });
    }

    get prevComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the first item return the last
        if (index - 1 < 0) {
            return this.componentsNavigationOrder[this.componentsNavigationOrder.length - 1];
        } else {
            return this.componentsNavigationOrder[index - 1];
        }
    }

    get nextComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the last item return the first
        if (index + 1 >= this.componentsNavigationOrder.length - 1) {
            return this.componentsNavigationOrder[0];
        } else {
            return this.componentsNavigationOrder[index + 1];
        }
    }

    getRouterIndex(tag: string): string {
        return this.$routerIndex.for(tag);
    }

    getCategoryComponents(category): ComponentMeta[] {
        return this.$meta.componentState[category];
    }

    urlMatchPath(categoryId: string) {
        let path: string = this.$i18n.translate(`categories:${categoryId}-route`);
        return this.fullPath.includes(`/${path}`);
    }

    onSideMenuSelection() {
        this.$refs.expandablePanel.onSideMenuSelection();
    }

    onPrevLinkClick() {
        (this.$refs.prevLink.$el as HTMLElement).blur();
    }

    onNextLinkClick() {
        (this.$refs.nextLink.$el as HTMLElement).blur();
    }

    @Watch('$route', { immediate: true })
    private routeChanged(): void {
        this.fullPath = this.$route.fullPath;
    }
}
