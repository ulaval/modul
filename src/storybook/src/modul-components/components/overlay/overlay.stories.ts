import { actions } from '@storybook/addon-actions';
import { boolean, text } from '@storybook/addon-knobs';
import { OVERLAY_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${OVERLAY_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    methods: actions(
        'save',
        'cancel'
    ),
    template: `<div>
            <${OVERLAY_NAME} title="Title" @save="save" @cancel="cancel">
                <m-button slot="trigger">Open</m-button>
                <p>
                    Lorem, ipsum dolor sit amet consectetur adipisicing elit.
                </p>
            </${OVERLAY_NAME}>
        </div>`
});

defaultStory.story = {
    name: 'default'
};

export const sandbox = () => ({
    methods: actions(
        'emitSave',
        'emitCancel',
        'emitOpen',
        'emitClose',
        'emitPortalAfterOpen',
        'emitPortalAfterClose'
    ),
    props: {
        id: {
            default: text('Prop id', `${OVERLAY_NAME}-stories-sandbox`)
        },
        open: {
            default: boolean('Prop open', true)
        },
        disabled: {
            default: boolean('Prop disabled', false)
        },
        className: {
            default: text('Prop class-name', 'class-name-test')
        },
        preload: {
            default: boolean('Prop preload', false)
        },
        lazy: {
            default: boolean('Prop lazy', true)
        },
        padding: {
            default: boolean('Prop padding', true)
        },
        paddingHeader: {
            default: boolean('Prop padding-header', true)
        },
        paddingBody: {
            default: boolean('Prop padding-body', true)
        },
        paddingFooter: {
            default: boolean('Prop padding-footer', true)
        },
        disableSaveButton: {
            default: boolean('Prop disable-save-button', false)
        },
        waiting: {
            default: boolean('Prop waiting', false)
        },
        focusManagement: {
            default: boolean('Prop focus-management', true)
        },
        slotDefault: {
            default: boolean('Slot default', true)
        },
        slotHeader: {
            default: boolean('Slot header', false)
        },
        slotFooter: {
            default: boolean('Slot footer', false)
        }
    },

    template: `<${OVERLAY_NAME}
        :id="id"
        :open="open"
        :disabled="disabled"
        :class-name="className"
        :preload="preload"
        :lazy="lazy"
        :padding="padding"
        :padding-header="paddingHeader"
        :padding-body="paddingBody"
        :padding-footer="paddingFooter"
        :disable-save-button="disableSaveButton"
        :waiting="waiting"
        :focus-management="focusManagement"
        @save="emitSave"
        @cancel="emitCancel"
        @open="emitOpen"
        @close="emitClose"
        @portal-after-open="emitPortalAfterOpen"
        @portal-after-close="emitPortalAfterClose"
    >
        <p v-if="slotDefault" class="m-u--no-margin">Slot default</p>
        <div v-if="slotHeader" slot="header">(Slot header)</div>
        <div v-if="slotFooter" slot="footer">(Slot footer)</div>
    </${OVERLAY_NAME}>`
});

export const overflowContent = () => ({
    template: `<div>
            <${OVERLAY_NAME} title="Title" >
                <m-button slot="trigger">Open</m-button>
                <h2 slot="header"
                style="margin:0;">Title</h2>
                <p>Native input</p>
                <input type="text" />
                <input type="text" />
                <p class="m-u--margin-top--l">modUL input</p>
                <p style="margin:0;">
                    <m-textfield label="Lorem ipsum dolor sit amet"></m-textfield>
                </p>
                <p style="margin:0;">
                    <m-textfield label="Lorem ipsum dolor sit amet"></m-textfield>
                </p>
                <p>
                    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
                    nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
                    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
                    nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
                    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
                    nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
                    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
                    nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
                    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
                    nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
                    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
                    nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
                    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
                    nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
                    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
                    nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
                </p>
            </${OVERLAY_NAME}>
        </div>`
});

export const loading = () => ({
    data: () => ({
        waiting: false
    }),
    methods: {
        toggleWaiting(): void {
            this.waiting = !this.waiting;
        }
    },
    template: `<div>
                <${OVERLAY_NAME} title="Titre"develop
                    :waiting="waiting">
                    <m-button slot="trigger">Open</m-button>
                    <h2 slot="header"
                    style="margin: 0;">Title</h2>

                    <m-button @click="toggleWaiting">Put waiting mode</m-button>

                </${OVERLAY_NAME}>
        </div>`
});

export const customFooter = () => ({
    methods: {
        fermerFenetre(): void {
            console.log('fermer');
        }
    },
    template: `<div>
    <${OVERLAY_NAME} title="Title" >
        <m-button slot="trigger">Open</m-button>
        <h2 slot="header" style="margin:0;">Custom footer</h2>
        <p>
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
        </p>
        <p>
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis magni reiciendis iste nulla eum velit error
            nihil corrupti. At culpa accusamus autem dignissimos aliquid quae ratione ipsa, reiciendis quod vero?
        </p>
        <div slot="footer">
            <m-link mode="button"
                    @click="fermerFenetre()"
                    ref="boutonFermer">
                fermer
            </m-link>
        </div>
    </${OVERLAY_NAME}>
</div>`
});

export const hideFooter = () => ({
    template: `<div>
    <${OVERLAY_NAME} :hide-footer="true" :open="true" >
        <h2 slot="header">Title</h2>
    </${OVERLAY_NAME}>
</div>`
});
