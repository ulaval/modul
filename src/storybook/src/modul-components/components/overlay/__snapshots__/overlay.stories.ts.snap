// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-overlay Custom Footer 1`] = `
<div>
  <div
    class="m-overlay"
    title="Title"
  >
    <button
      class="m-button m--is-skin-primary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Open 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots modul-components|m-overlay Loading 1`] = `
<div>
  <div
    class="m-overlay"
    title="Title"
  >
    <button
      class="m-button m--is-skin-primary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Open 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots modul-components|m-overlay Overflow Content 1`] = `
<div>
  <div
    class="m-overlay"
    title="Title"
  >
    <button
      class="m-button m--is-skin-primary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Open 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots modul-components|m-overlay default 1`] = `
<div>
  <div
    class="m-overlay"
    title="Title"
  >
    <button
      class="m-button m--is-skin-primary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        Open 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <!---->
  </div>
</div>
`;
